### PR TITLE
Refactor entity generator file names from came case to spinal case

### LINKED ISSUE #2753: Follow dashed naming convention for entities
Entity file names and states are camelCased instead of dashed cased. This does not follow the naming convention in jhipster. John Papa also uses dashed-cased in his examples.

While we are breaking things with the 3.0 release we should fix this as well. 

I can work on this.

### PR DESCRIPTION
**Pull request by andidev**
fix #2753

### CODE REVIEW

**Comment by andidev** [comment]
- renamed templates to match
- refactored entity filenames, states, urls to be spinal-cased (dashed) instead of cameCased
- refactored to use specific in-memory variables for directoryname, filename, state, url, etc

please merge this as soon as possible to avoid nasty confilcts

File: entity/index.js [hunk 1]
```diff
@@ -1447,12 +1451,31 @@ module.exports = EntityGenerator.extend({
...
+            var entityNamePluralizedAndSpinalCased = _s.trim(_s.dasherize(pluralize(this.name)), '-');
+
+            this.entityApiUrl = entityNamePluralizedAndSpinalCased;
             this.entityClass = _s.capitalize(this.name);
             this.entityClassPlural = pluralize(this.entityClass);
             this.entityInstance = _s.decapitalize(this.name);
             this.entityInstancePlural = pluralize(this.entityInstance);
             this.entityTableName = _s.underscored(this.name).toLowerCase();
 
+            this.entityFolderName = entityNameSpinalCased + '-management';
```
Comment (andidev) [inline entity/index.js]: Maybe the folder name should exclude the '-management' suffix?

Comment (deepu105) [inline entity/index.js]: Yes

**Comment by deepu105** [comment]
Looks good, but i haven't tested

File: entity/index.js [hunk 2]
```diff
@@ -1447,12 +1451,31 @@ module.exports = EntityGenerator.extend({
             } else {
                 this.pkType = 'Long';
             }
+
+            var entityNameSpinalCased = _s.trim(_s.dasherize(this.name), '-');
+            var entityNamePluralizedAndSpinalCased = _s.trim(_s.dasherize(pluralize(this.name)), '-');
```
Comment (cbornet) [inline entity/index.js]: I think decapitalizing before dasherizing would be nicer

Comment (andidev) [inline entity/index.js]: Yes that's true and make more sense

File: entity/index.js [hunk 3]
```diff
@@ -1447,12 +1451,31 @@ module.exports = EntityGenerator.extend({
...
+
+            this.entityApiUrl = entityNamePluralizedAndSpinalCased;
```
Comment (andidev) [inline entity/index.js]: What do you think of the entity prefix for for all in-memory variables. Should we remove them? The problem with that is that they could overwrite app properties. 
An option would be to put them in an object defined as 'e'  or 'entity' though that will be more verbose. 
I'm a bit unsure about this. Any suggestions?

**Comment by andidev** [comment]
Maybe we should read the added in-memory variables from the entity json file as well and default to the current names. This would make it really customizable. Though maybe it is overkill.

**Comment by andidev** [comment]
Tested with mysql with complex relationships + mongodb so I would say this is ready to be merged.
Change could be further polished but that can be done in a later PR if needed.

**Pull request merged**